resource_alks_iamrole.go

```diff
@@ -192,56 +184,39 @@ func resourceAlksIamTrustRoleCreate(d *schema.ResourceData, meta interface{}) er
 
 	log.Printf("[INFO] alks_iamtrustrole.id: %v", d.Id())
 
-	return resourceAlksIamRoleRead(d, meta)
+	return diags
```

Comment:
Since `diags` is only defined and not set anywhere, won't this just be `nil`? If that's the case it would probably make it a little clearer to just directly return `nil` so that it doesn't appear like we're returning a non-nil object

Reply:
Great find here. You're sorta correct, but basically - on a CREATE you always want to return a READ so that Terraform can set the ID and so that the object can be tracked in the state. This function for IAM Trust Role was incorrectly set to return `diags`, so that's my bad. Changing this now!